final_project.py

```diff
@@ -0,0 +1,369 @@
+# as well as to implement a Make-Your-Own procedure (MYOP). You are encouraged 
+# to define any additional helper procedures that can assist you in accomplishing 
+# a task. You are encouraged to test your code by using print statements and the 
+# Test Run button. 
+# ----------------------------------------------------------------------------- 
+
+# Example string input. Use it to test your code.
+example_input="John is connected to Bryant, Debra, Walter.\
+John likes to play The Movie: The Game, The Legend of Corgi, Dinosaur Diner.\
+Bryant is connected to Olive, Ollie, Freda, Mercedes.\
+Bryant likes to play City Comptroller: The Fiscal Dilemma, Super Mushroom Man.\
+Mercedes is connected to Walter, Robin, Bryant.\
+Mercedes likes to play The Legend of Corgi, Pirates in Java Island, Seahorse Adventures.\
+Olive is connected to John, Ollie.\
+Olive likes to play The Legend of Corgi, Starfleet Commander.\
+Debra is connected to Walter, Levi, Jennie, Robin.\
+Debra likes to play Seven Schemers, Pirates in Java Island, Dwarves and Swords.\
+Walter is connected to John, Levi, Bryant.\
+Walter likes to play Seahorse Adventures, Ninja Hamsters, Super Mushroom Man.\
+Levi is connected to Ollie, John, Walter.\
+Levi likes to play The Legend of Corgi, Seven Schemers, City Comptroller: The Fiscal Dilemma.\
+Ollie is connected to Mercedes, Freda, Bryant.\
+Ollie likes to play Call of Arms, Dwarves and Swords, The Movie: The Game.\
+Jennie is connected to Levi, John, Freda, Robin.\
+Jennie likes to play Super Mushroom Man, Dinosaur Diner, Call of Arms.\
+Robin is connected to Ollie.\
+Robin likes to play Call of Arms, Dwarves and Swords.\
+Freda is connected to Olive, John, Debra.\
+Freda likes to play Starfleet Commander, Ninja Hamsters, Seahorse Adventures."
+
+# ----------------------------------------------------------------------------- 
+# create_data_structure(string_input): 
+#   Parses a block of text (such as the one above) and stores relevant 
+#   information into a data structure. You are free to choose and design any 
+#   data structure you would like to use to manage the information.
+# 
+# Arguments: 
+#   string_input: block of text containing the network information
+#
+#   You may assume that for all the test cases we will use, you will be given the 
+#   connections and games liked for all users listed on the right-hand side of an
+#   'is connected to' statement. For example, we will not use the string 
+#   "A is connected to B.A likes to play X, Y, Z.C is connected to A.C likes to play X."
+#   as a test case for create_data_structure because the string does not 
+#   list B's connections or liked games.
+#   
+#   The procedure should be able to handle an empty string (the string '') as input, in
+#   which case it should return a network with no users.
+# 
+# Return:
+#   The newly created network data structure
+def create_data_structure(s):
+    network = {} # {user1:[[connections],[games]], user2:[[connections],[games]]...} 
```

Comment:
this data structure works, but a slightly better way to organize would be 2 dictionaries: one for connections & one for games. then you don't need a list of lists as values. it will be 2 dictionaries where the values are lists